File: src/main/java/pvz/community/PlantsVSZombies.java
package pvz.community;

import lombok.Getter;
import pvz.community.launcher.LauncherSettings;
import pvz.community.renderer.WindowObjectRenderer;
import pvz.community.renderer.window.Window;
import pvz.community.utils.UpTime;

public class PlantsVSZombies {

    @Getter
    private static PlantsVSZombies instance = null;
    @Getter
    private final LauncherSettings settings;
    @Getter
    private final Window window;
    @Getter
    private final UpTime systemUtils;
    @Getter
    private final WindowObjectRenderer renderer;

    public PlantsVSZombies(LauncherSettings settings) {
        instance = this;
        this.settings = settings;
        this.systemUtils = new UpTime();
        this.window = Window.get();
        window.run();
        this.renderer = new WindowObjectRenderer();
    }
}

File: src/main/java/pvz/community/PlantsVSZombies.kt
package pvz.community

import lombok.Getter
import pvz.community.launcher.GameLauncher
import pvz.community.launcher.LauncherSettings
import pvz.community.renderer.window.GameWindow

/**
    This class is the main game instance.
    It is responsible for starting the game, and for handling the game loop.
    It takes LauncherSettings as a parameter, which is used to configure the game.
    Launcher Settings are stored in the properties.json file
    @see GameLauncher
 **/
class PlantsVSZombies(settings: LauncherSettings){

    private var s = settings;
    fun getSettings(): LauncherSettings {
        return s;
    }
    @Getter private var window: GameWindow? = GameWindow();


    // Create a method to access this class from anywhere in the game
    companion object {
        fun getInstance(): GameLauncher? {
            return GameLauncher.getInstance();
        }
    }








}